Runtime/Scripts/Types/RGClientConnection_Remote.cs

```diff
@@ -61,13 +66,40 @@ private bool SendSocketMessage(string type, string data)
                 // put the length header into the buffer first
                 BinaryPrimitives.WriteInt32BigEndian(finalBuffer, dataBuffer.Length);
                 Array.Copy(dataBuffer, 0, finalBuffer, 4, dataBuffer.Length);
-                var vt = _client.GetStream().WriteAsync(finalBuffer);
-                vt.ConfigureAwait(false).GetAwaiter().OnCompleted(() =>
+                var canWrite = _client?.Connected == true && _client?.Client.Poll(1, SelectMode.SelectWrite) == true;
+                if (canWrite)
+                {
+                    var vt = _client.GetStream().WriteAsync(finalBuffer);
+                    vt.ConfigureAwait(false).GetAwaiter().OnCompleted(() =>
+                    {
+                        // see if the socket is no longer writeable
+                        if (!vt.IsCompletedSuccessfully)
+                        {
+                            HandleDeadWriteState();
+                        }
+                    });
```

Comment:
is there a need to implement a retry policy if this write fails?

Reply:
updates are not guaranteed to get there.  If the connection is down, all the states between 'now' and 'future now' are not forwarded to the client as they should have no need to process them anyway

in fact.. on the client side if it is processing slowly and we get multiple new states before we finish processing old states, we also explicitly trash the states in between